cranelift/codegen/src/machinst/abi_impl.rs

```diff
@@ -1150,6 +1169,28 @@ impl<M: ABIMachineSpec> ABICallee for ABICalleeImpl<M> {
                     ));
                 }
             }
+            &ABIArg::ImplicitArg { pointer, ty, .. } => {
+                let into_reg = into_regs.only_reg().unwrap();
+                // We need to dereference the pointer.
+                let base = match &pointer {
+                    &ABIArgSlot::Reg { reg, .. } => Reg::from(reg),
+                    &ABIArgSlot::Stack { offset, ty, .. } => {
+                        // In this case we need a temp register to hold the address.
+                        // Use the stack limit register as temp.
```

Comment:
As an alternative I suppose we could provide an "allocate temp vreg" callback to `gen_copy_arg_to_regs`; this codegen happens before regalloc so it is still an option to create vregs here. Perhaps a comment here to record the altnernate possibility? (We may in the future want to remove the need for more dedicated non-allocatable registers, at which point it may become more relevant.)

Reply:
Ah yes, that's better.   However, I wasn't able to get a callback past the borrow checker - maybe I'm just not experienced enough with Rust, but it does look a real problem:
```
                for insn in self.vcode.abi().gen_copy_arg_to_regs(i, regs).into_iter() {
```
This already holds a borrow on `self` during the `gen_copy_arg_to_regs` call.  Passing a closure would require another (mutable) borrow on `self`, which isn't allowed.

Instead, I noticed that there is already a mechanism to pass a temp reg to `ABICallee` via the `temp_needed`/`init` mechanism, so I simply extended that to support multiple temps.